service/controller/drainer.go

```diff
@@ -31,41 +30,29 @@ func NewDrainer(config DrainerConfig) (*Drainer, error) {
 
 	var err error
 
-	var v1ResourceSet *controller.ResourceSet
+	var resourceSet []resource.Interface
 	{
-		c := v1.DrainerResourceSetConfig{
-			K8sClient: config.K8sClient,
-			Logger:    config.Logger,
-		}
-
-		v1ResourceSet, err = v1.NewDrainerResourceSet(c)
+		resourceSet, err = NewDrainerResourceSet(DrainerResourceSetConfig(config))
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
 	}
 
-	var v2ResourceSet *controller.ResourceSet
+	var operatorkitController *controller.Controller
 	{
-		c := v2.DrainerResourceSetConfig{
-			K8sClient: config.K8sClient,
-			Logger:    config.Logger,
+		resourceSets := [][]resource.Interface{
+			resourceSet,
 		}
 
-		v2ResourceSet, err = v2.NewDrainerResourceSet(c)
-		if err != nil {
-			return nil, microerror.Mask(err)
+		resources := []resource.Interface{}
+		for _, set := range resourceSets {
+			resources = append(resources, set...)
 		}
-	}
 
-	var operatorkitController *controller.Controller
-	{
 		c := controller.Config{
-			K8sClient: config.K8sClient,
-			Logger:    config.Logger,
-			ResourceSets: []*controller.ResourceSet{
-				v1ResourceSet,
-				v2ResourceSet,
-			},
+			K8sClient:    config.K8sClient,
+			Logger:       config.Logger,
+			Resources:    resources,
 			ResyncPeriod: 2 * time.Minute,
 			NewRuntimeObjectFunc: func() runtime.Object {
 				return new(v1alpha1.DrainerConfig)
```

Comment:
Do these already have the correct labels? Would you keep the existing operator around as a `thiccc` operator and then release this as a "v3" or something? Just thinking about how it's going to be applied to resources and versioned.

Reply:
That's a good point. I haven't done so much with `thiccc` operators, but I realize now that previously this was reconciling based on the version bundle version which is now gone (and different than the tagged project version). 
`v1` previously reconciled `0.1.0` Drainerconfigs.
`v2` previously reconciled `0.2.0` Drainerconfigs. This is the current behavior.

Going back to AWS operator 5.5.0 (and even `v25` on `thiccc`) the DrainerConfigs are already version `0.2.0`, so I wouldn't _think_ there are any `v1`s still floating around.

The current behavior has been reconciling those 0.2.0 resources, but do I need to do anything to version them for the future? cc @MarcelMue 
(See also my comment about `handlesFunc` and `initCtx` in new `operatorkit`)

Reply:
I think this is why we need more refactoring here:
Currently just the folder structure was flattened and some parts moved.
But parts are missing - e.g. moving the version information into the correct place (example):
https://github.com/giantswarm/release-operator/blob/1c13cd0d07953ac8bd4a5a9d520b2eec7c20d2c8/pkg/project/project.go#L8

And then we need to figure out how we deal with the version correctly. The main problem is, that this operator is only running once - not multiple times currently. We should have a bigger discussion what the future for the operator is. To me it makes sense to "hack it" and flatten it but hardcode it to version 0.2.0 for now. Otherwise we need to make a whole transition to 0.3.0 or something like that.

Reply:
It has a version, but operatorkit doesn't seem to care about it. How do I tell operatorkit to reconcile a certain version without handlesFunc? DrainerConfigs also don't get created here, so if we want to apply labels to them, we have to change that in provider operators

Reply:
See the selector here for reference:
https://github.com/giantswarm/aws-operator/blob/deef2c9376b744141cdb2c53ab9161ab42ee1f04/service/controller/machine_deployment.go#L137-L139

Reply:
Existing CRs don't have labels, so we can decide on a name for one going forward. Let's say `node-operator.giantswarm.io/version`.

How about this:
1. set a `Selector` in this or a followup PR which returns `false` if _any_ version is set for that label
2. release it to handle 0.2.0 CRs

Then, for the future:
3. set a `Selector` to match the actual operator version, like elsewhere
4. deploy `node-operator` as a release component
5. modify provider operators to attach the version label to new CRs